proxy/logging/LogConfig.cc

```diff
@@ -765,13 +775,21 @@ LogConfig::update_space_used()
     if (sret != -1 && S_ISREG(sbuf.st_mode)) {
       total_space_used += (int64_t)sbuf.st_size;
 
-      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name) && candidate_count < MAX_CANDIDATES) {
+      if (auto_delete_rolled_files && LogFile::rolled_logfile(entry->d_name)) {
         //
-        // then add this entry to the candidate list
+        // then check if the candidate belongs to any given log type
         //
-        candidates[candidate_count].name  = ats_strdup(path);
-        candidates[candidate_count].size  = (int64_t)sbuf.st_size;
-        candidates[candidate_count].mtime = sbuf.st_mtime;
+        ts::TextView type_name(entry->d_name, strlen(entry->d_name));
+        auto suffix = type_name;
+        type_name.remove_suffix(suffix.remove_prefix(suffix.find('.') + 1).remove_prefix(suffix.find('.')).size());
+        auto iter = deleting_info.find(type_name);
+        if (iter == deleting_info.end()) {
+          // We won't delete the log if its name doesn't match any give type.
+          break;
+        }
+
+        auto &candidates = iter->candidates;
+        candidates.push_back(LogDeleteCandidate(path, (int64_t)sbuf.st_size, sbuf.st_mtime));
         candidate_count++;
```

Comment:
Is this needed? Why doesn't `candidates.size()` suffice?

Reply:
This is the total count of all `candidates` array.

Reply:
How is that different from `candidates.size()`?

Reply:
There are multiple `candidates`, one for each log type.